### PR TITLE
D8CORE-457 and more

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Lots of configuration settings and a new install module
- Moved config to config/sync due to config_installer issue
- Missing all sorts of tests

# Needed By (Date)
- ?

# Urgency
- Semi important

# Steps to Test

1. See steps 1-4 in https://github.com/SU-SWS/acsf-cardinalsites/pull/18
5. D8CORE-101: Review that the main nav is populated with items after install
6. D8CORE-148: Review that the installation has default content
7. D8CORE-148: Review that the default content is in a layout_builder template with sidenav (home page should be different but there is a missing feature in layout_builder so I've created a new ticket to look in to a work around.)
8. D8CORE-151: Review user roles and Howard/Lindsey's grants.
9. D8CORE-458: Review Card paragraph type for construction and aesthetics. 
10. D8CORE-456: Review banner paragraph type for construction and aesthetics
11. Review the rest of the configuration settings and code.

# Affected Projects or Products
- Lots

# Associated Issues and/or People
- D8CORE-101
- D8CORE-148
- D8CORE-151
- D8CORE-458
- D8CORE-457
- D8CORE-456

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
This seems to work. but some undesired changes/deletions have occured. I'll fix them up

**Comment by pookmish** [comment]
This is going to rely on https://github.com/SU-SWS/stanford_fields/pull/1 to fix up the component field displays. Overall this looks goo though.

**Comment by pookmish** [comment]
@boznik I've taken what shea started, i've added a few things and restored some stuff. The overall aproach he was going with is the same. The links in the cards and banners I want to review next week. but can you have a look over to see if anything stands out to you

**Comment by pookmish** [comment]
This work is visible on https://mikedeleteme20190903-dev.sites.stanford.edu/

File: composer.json [hunk 1]
```diff
@@ -15,19 +15,29 @@
         "drupal/address": "~1.2",
         "drupal/admin_toolbar": "~1.20",
         "drupal/allowed_formats": "~1.1",
+        "drupal/better_normalizers": "1.x-dev",
```
Comment (sherakama) [inline composer.json]: Lets lock this to a hash

File: composer.json [hunk 2]
```diff
@@ -15,19 +15,29 @@
...
         "drupal/google_analytics": "~2.1",
         "drupal/honeypot": "^1.27",
-        "drupal/imagemagick": "^3.0",
-        "drupal/menu_block": "^1.5",
+        "drupal/imagemagick": "~3.0@alpha",
+        "drupal/layout_builder_modal": "~1.0@alpha",
+        "drupal/layout_builder_restrictions": "~2.2",
+        "drupal/layout_library": "~1.0@beta",
+        "drupal/menu_admin_per_menu": "^1.0",
+        "drupal/menu_block": "dev-1.x",
```
Comment (sherakama) [inline composer.json]: What was wrong with version 1.5? Can we lock this to a hash?

File: composer.json (hunk 1 above)
Comment (pookmish) [reply composer.json]: i dont know why it decided to use that version. i can switch it to the beta

File: composer.json (hunk 2 above)
Comment (pookmish) [reply composer.json]: it doesnt have the "follow active parent" https://www.drupal.org/project/menu_block/issues/2756675
We might use it, we might not. :man_shrugging:

**Pull request merged**